runtime/tests/checker/errors_test.go

```diff
@@ -107,3 +108,83 @@ func TestCheckErrorShortCircuiting(t *testing.T) {
 		assert.IsType(t, &sema.NotDeclaredError{}, errs[0])
 	})
 }
+
+func TestCheckEntitlementsErrorMessage(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("reference subtyping", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+				access(all) entitlement E 
+				fun foo() {
+					let x = 1
+					let refX: auth(E) &Int = &x as auth(F) &Int
+				}
+            `,
+		)
+
+		errs := RequireCheckerErrors(t, err, 2)
+
+		require.IsType(t, &sema.NotDeclaredError{}, errs[0])
+		require.IsType(t, &sema.TypeMismatchError{}, errs[1])
+		require.Equal(t, "expected `auth(E) &Int`, got `auth(<<INVALID>>) &Int`", errs[1].(*sema.TypeMismatchError).SecondaryError())
+	})
+
+	t.Run("invalid access", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+				access(all) entitlement E 
+				access(all) resource R {
+					access(E) fun foo() {}
+				}
+				fun foo() {
+					let r <- create R() 
+					let refR = &r as auth(F) &R
+					refR.foo()
+					destroy r
+				}
+            `,
+		)
+
+		errs := RequireCheckerErrors(t, err, 2)
+
+		require.IsType(t, &sema.NotDeclaredError{}, errs[0])
+		require.IsType(t, &sema.InvalidAccessError{}, errs[1])
```

Comment:
Maybe use `require.ErrorAs` here to cast/assert in one place, and avoid the cast below.
Same for the other sub-test